flight/tests/misc_math/unittest.cpp

```diff
@@ -237,3 +237,63 @@ TEST_F(CircularModulusDeg, SweepError) {
     ASSERT_NEAR(error, circular_modulus_deg(error + 1080), eps);
   }
 };
+
+// Test fixture for linear_interpolate()
+class LinearInterpolate : public MiscMath {
+protected:
+  virtual void SetUp() {
+  }
+
+  virtual void TearDown() {
+  }
+};
+
+TEST_F(LinearInterpolate, ThrottleCurve1to1) {
+  float const range_min = 0.0f;
+  float const range_max = 1.0f;
+  uint8_t const curve_numpts = 5;
+  float const curve[5] = { 0.0f, 0.25f, 0.5f, 0.75f, 1.0f };
+
+  // 21 points in range
+  for (size_t i = 0; i <= 20; ++i) {
+    float const input = i * 0.05f;
+    EXPECT_EQ(input, linear_interpolate(input, curve, curve_numpts, range_min, range_max));
+  }
+
+  // 10 points below min range
+  for (size_t i = 1; i <= 10; ++i) {
+    float const input = range_min - i * 0.1f;
```

Comment:
This isn't really an issue because you do it in the above tests, but it doesn't explicitly test the behavior at the boundary. If you do another commit, it might be good to test that ```range_max``` maps to ```range_max```, in case the first unit test gets changed.